scripts/lib/zulip_tools.py

```diff
@@ -198,13 +198,23 @@ def run(args, **kwargs):
     try:
         subprocess.check_call(args, **kwargs)
     except subprocess.CalledProcessError:
+        subcommand = " ".join(map(shlex.quote, args))
         print()
         print(WHITEONRED + "Error running a subcommand of %s: %s" %
-              (sys.argv[0], " ".join(map(shlex.quote, args))) +
+              (sys.argv[0], subcommand,) +
               ENDC)
         print(WHITEONRED + "Actual error output for the subcommand is just above this." +
               ENDC)
         print()
+        # Yarn breaking `/tools/provision` is common enough that giving a hint here can
+        # really help when debugging a failure.  Yarn will also keep making reattempts
+        # with a spotty network connection.  Instead, we should just print the message
+        # below and exit.
+        if get_environment() == 'dev' and 'yarn' in subcommand:
+            print("Failures involving Yarn are usually indicative of a network issue!\n"
+                  "Please make sure you have a reliable network connection before reattempting the "
+                  "provisioning process.")
+            sys.exit(1)
```

Comment:
This check doesn’t belong in the general-purpose `run` routine. It belongs in an exception handler in the particular caller that you’re targeting. Which in this case is [here in provision.py](https://github.com/zulip/zulip/blob/2.0.4/tools/lib/provision.py#L377-L387).

Reply:
Thank you for pointing this out, Anders.